src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -177,61 +167,80 @@ def _create_urls(self, purl: PackageURL) -> tuple[list[str], RepoFinderInfo]:
+            purl=False, path="/".join(["systems", purl.type, "packages", purl.name, "versions", purl.version])
+        )
+        target_url = urllib.parse.urlunsplit(api_endpoint)
+
+        result = send_get_http(target_url, headers={})
+        if not result:
+            return None, False
+
+        result_attestations = json_extract(result, ["attestations"], list)
+        if not result_attestations:
+            logger.debug("No attestations in result.")
+            return None, False
+        if len(result_attestations) > 1:
+            logger.debug("More than one attestation in result: %s", len(result_attestations))
+
+        attestation_url = json_extract(result_attestations, [0, "url"], str)
+        if not attestation_url:
+            logger.debug("No attestation reported for %s", purl)
+            return None, False
+
+        attestation_data = send_get_http(attestation_url, headers={})
+        if not attestation_data:
+            return None, False
+
+        bundle = json_extract(attestation_data, ["attestation_bundles"], list)
+        if not bundle:
+            logger.debug("No attestation bundle in response.")
+            return None, False
+        if len(bundle) > 1:
+            logger.debug("Bundle length greater than one: %s", len(bundle))
+
+        attestations = json_extract(bundle[0], ["attestations"], list)
+        if not attestations:
+            logger.debug("No attestations in response.")
+            return None, False
+        if len(attestations) > 1:
+            logger.debug("More than one attestation: %s", len(attestations))
+
+        if not isinstance(attestations[0], dict):
+            logger.debug("Attestation invalid.")
+            return None, False
+
+        return attestations[0], json_extract(result_attestations, [0, "verified"], bool) or False
 
-    def _read_json(self, json_data: str) -> tuple[list[str], RepoFinderInfo]:
+    @staticmethod
+    def extract_links(json_data: dict) -> tuple[list[str], RepoFinderInfo]:
         """
-        Parse the deps.dev json file and extract the repository links.
+        Extract the repository links from  the deps.dev json data.
 
         Parameters
         ----------
-        json_data : str
-            The json metadata as a string.
+        json_data : dict
+            The json metadata.
 
         Returns
         -------
         tuple[list[str], RepoFinderOutcome] :
             The extracted contents as a list, and the outcome to report.
         """
-        try:
-            parsed = json.loads(json_data)
-        except ValueError as error:
-            logger.debug("Failed to parse response from deps.dev: %s", error)
-            return [], RepoFinderInfo.DDEV_JSON_FETCH_ERROR
-
-        links_keys = ["version", "links"] if "version" in parsed else ["links"]
-        links = json_extract(parsed, links_keys, list)
+        links_keys = ["version", "links"] if "version" in json_data else ["links"]
```

Comment:
I'm curious why there is some different in where the links are stored in the JSON data? Is that because different endpoints returns different schema or depending on the package we are searching 🤔 ?

Reply:
As above, yes.